### PR TITLE
Separate find dependencies in layouts for incidents and indicators

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:  https://github.com/demisto/etc/issues/35366

## Description
added an if statement to check the kind of the layout to search dependencies by type (incidents, indicators)

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/tests/dependencies_test.py [hunk 1]
```diff
@@ -884,11 +884,15 @@ def create_inputs_for_method(repo, current_pack, inputs_arguments):
                 pack_to_take_entity_from = random.choice(range(1, number_of_packs))
                 input_argument.append(get_entity_by_pack_number_and_entity_type(repo, pack_to_take_entity_from,
                                                                                 LIST_ARGUMENTS_TO_METHODS[arg_type]))
+                if arg_type == 'indicators_fields':
+                    continue
```
Comment (MosheEichler) [inline demisto_sdk/commands/common/tests/dependencies_test.py:888]: Because the layout is of type incident

Comment (reutshal) [reply demisto_sdk/commands/common/tests/dependencies_test.py:888]: @MosheEichler Please add a comment here and in the next condition explaining the logic

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 2]
```diff
@@ -218,15 +218,17 @@ def _search_packs_by_items_names(items_names: Union[str, list],
...
         """
         Searches for implemented packs of the given items.
 
         Args:
             items_names (str or list): items names to search.
             items_list (list): specific section of id set.
             exclude_ignored_dependencies (bool): Determines whether to include unsupported dependencies or not.
-
+            incident_or_indicator (str): 'indicator' if should focus on indicator fields,
+                 'incident' if should focus on incident fields and None to focus on both.
```
Comment (reutshal) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:231]: Why not 'both' in the last case? 
And please improve the description here, it is not clear what 'focus' means

File: demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py [hunk 3]
```diff
@@ -881,19 +882,237 @@ def test_collect_playbooks_dependencies_skip_unavailable(self, id_set):
...
+            "group": 0,
+            "hidden": "false",
+            "associatedToAll": "true",
+            "unmapped": "false",
+            "unsearchable": "false",
+            "caseInsensitive": "true",
+            "sla": 0,
+            "threshold": 72,
+            "fromVersion": "5.5.0"
+        }
```
Comment (reutshal) [inline demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py:932]: Do we really need all of these fields in the entities?

File: demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py [hunk 4]
```diff
@@ -881,19 +882,237 @@ def test_collect_playbooks_dependencies_skip_unavailable(self, id_set):
...
 
         When
             - Building dependency graph for pack.
 
         Then
             - Extracting the packs that the layout depends on.
         """
-        expected_result = {("FeedMitreAttack", True), ("PrismaCloudCompute", True), ("CommonTypes", True),
-                           ("CrisisManagement", True)}
+        expected_result = {("PrismaCloudCompute", True)}
```
Comment (reutshal) [inline demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py:1115]: Why did you change this test?

**Review by reutshal — CHANGES_REQUESTED** [review]
Nice usage in the TestSuite!
@roysagi Would be great if you could also take a look

File: demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py (hunk 4 above)
Comment (MosheEichler) [reply demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py:1012]: Because the rest of the packs has indicator_fields and the layout is of type incident, so the layout is not depend on them

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 5]
```diff
@@ -218,15 +218,19 @@ def _search_packs_by_items_names(items_names: Union[str, list],
     @staticmethod
     def _search_packs_by_items_names_or_ids(items_names: Union[str, list],
                                             items_list: list,
-                                            exclude_ignored_dependencies: bool = True) -> set:
+                                            exclude_ignored_dependencies: bool = True,
+                                            incident_or_indicator: Optional[str] = 'both') -> set:
```
Comment (roysagi) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:222]: I think you are not passing this arg the calls in `_collect_layouts_dependencies` so either you should pass it or this arg is useless and should be removed

Comment (MosheEichler) [reply demisto_sdk/commands/find_dependencies/find_dependencies.py:222]: I added it

**Comment by reutshal** [comment]
@roysagi @amshamah419 Can someone approve the changelog? Thanks

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 6]
```diff
@@ -218,15 +218,19 @@ def _search_packs_by_items_names(items_names: Union[str, list],
...
         """
         Searches for implemented packs of the given items.
 
         Args:
             items_names (str or list): items names to search.
             items_list (list): specific section of id set.
             exclude_ignored_dependencies (bool): Determines whether to include unsupported dependencies or not.
-
+            incident_or_indicator (str):
+                'indicator' to search packs with indicator fields,
```
Comment (reutshal) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:231]: In the condition you are checking for "Indicator" / "Incident" / "both" (Camal cased). Please update the docstring accordingly.
I also think you should turn "both" into "Both" for consistency.

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 7]
```diff
@@ -236,11 +240,18 @@ def _search_packs_by_items_names_or_ids(items_names: Union[str, list],
...
+                item_possible_ids = [item_name, f'indicator_{item_name}', f'{item_name}-mapper']
+            elif incident_or_indicator == 'both':
+                item_possible_ids = [item_name, f'incident_{item_name}', f'indicator_{item_name}',
+                                     f'{item_name}-mapper']
+
             for item_from_id_set in items_list:
                 machine_name = list(item_from_id_set.keys())[0]
                 item_details = list(item_from_id_set.values())[0]
-                if (machine_name in item_possible_names or item_name == item_details.get('name')) \
+                if (machine_name in item_possible_ids or item_name == item_details.get('name')) \
```
Comment (reutshal) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:254]: Why did we change this?

**Review by reutshal — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/find_dependencies/find_dependencies.py (hunk 7 above)
Comment (MosheEichler) [reply demisto_sdk/commands/find_dependencies/find_dependencies.py:254]: https://github.com/demisto/demisto-sdk/pull/1290#discussion_r650486686

**Review by reutshal — APPROVED** [review]

**Review by reutshal — CHANGES_REQUESTED** [review]
Please move the RN to the correct changelog entry

**Review by reutshal — APPROVED** [review]
